src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    function notNullOrUndefined(element) {
+      return element !== null && element !== undefined;
+    }
```

Comment:
1. it is an independent function, so move it out of current function
2. You don't need to create that already exist. There is an operator as `??` in javascript. See comments bellow
3.